### ISSUE TITLE
Help about Expanded UITableView in a UITableViewCell

### ISSUE THREAD

**Issue by rjcristy** [smileyborg/TableViewCellWithAutoLayout#20]
First of all I want to send my sincerest thanks for this very concise Library (PureLayout) and also this wonderful example. I depend my project's most module on this example. 

I would like to ask how am I going to solve this problem. So I have a tableview (T2) in a tableViewCell (C1) in a tableView (T1) ; all composed in Purelayout. However, I cannot *always* get the total height of the Table (T2) so I could expand it fixed on C1.The current way that I get the total height of T2 is

    [self.commentContainer layoutIfNeeded];
    return [self.commentContainer contentSize].height 

I get the appropriate total height of T2 if every T2's cell has a fixed height. That's is, when a label in the T2 is only one line. The problem starts when the label has more than one line (and the NumberOfLines is set to 0 , meant to expand as the label's text becomes long ). 

In summary, I want to achieve something like a Youtube Comment Form. When I press the [show all Comments] Button, It shows the comments in non-scrollable way. I want to expand T2 most so it's contents won't scroll since T1 is already scrollable.

I created an illustration in hopes of helping myself explain better.

![screen shot 2015-03-26 at 9 55 59 pm](https://cloud.githubusercontent.com/assets/8848965/6847629/fa9a6d86-d402-11e4-99a8-3b888e3cc9a9.png)

Thank you very much smileyborg!

**Comment by smileyborg** [comment]
Hi @rjcristy,

I understand what you're trying to do. I know that nesting table views inside the cells of other table views can be difficult though, so this may not be easy.

My suggestion on how to make sure that you are always getting the correct height for T2 is to observe the `contentSize` using KVO:

Add Observer:
```
static void * const kObservingContentSizeChangesContext = (void *)& kObservingContentSizeChangesContext;

[tv addObserver:self forKeyPath:@"contentSize" options:NSKeyValueObservingOptionNew | NSKeyValueObservingOptionPrior context:&kObservingContentSizeChangesContext];
```

KVO Method:
```
- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context {
    if (context == &kObservingContentSizeChangesContext) {
        UITableView *tableView = object;
        NSLog(@"%@ contentSize changed to %@", tableView, NSStringFromCGSize(tableView.contentSize));
        self.heightConstraint.constant = tableView.contentSize.height;
    }
}
```

Don’t forget to remove yourself when done:
```
- (void)dealloc {
    [self.tableView removeObserver:self forKeyPath:@"contentSize" context:&kObservingContentSizeChangesContext];
}
```

Now, you'll notice that what we are doing is keeping an auto layout constraint that controls the height of the table view (T2), referenced as `self.heightConstraint` above. This can be created initially with any value, something like:

```
self.heightConstraint = [tv2 autoSetDimension:ALDimensionHeight toSize:44.0];
```

Then, we update the height of the table view whenever its contents change by setting the new `contentSize.height` to the `constant` of that constraint.

In order for T1 to know that it now must resize the cell (C1) because T2 has changed size, use:

```
[tv1 beginUpdates];
[tv1 endUpdates];
```

Finally, if you are finding that the `contentSize.height` of T2 is incorrect due to multi-line labels inside its cell(s), then the issue is within the cells of T2. So you should start by making sure that T2's cell heights are calculated correctly -- with multi-line labels, for example, you may need to manually set the `label.preferredMaxLayoutWidth` to the width of the label so that it word wraps correctly.

Hope that helps!

**Comment by rjcristy** [comment]
I'm very impressed by your suggestion. I learnt just now that technology. Thanks, @smileyborg  you saved me . More power!
